### PR TITLE
Unified config management and parameter logging: implementation before integration

### LINKED ISSUE #554: All exporters should use unified configuration management, and log their configuration
We have some manual logging, but this will help with debugging greatly. The lack of this made solving #553 really hard.

Unifying this would make sure we don't miss any new configuration that gets added. It's also better for security-- if we leave everything up to manual logging, someone could accidentally log credentials!

The way exporters are configured has always been inconsistent, and sometimes messy. A cleanup of this whole area would be valuable.

### PR DESCRIPTION
**Pull request by KevinMGranger**
Relates to #554.

This adds the module responsible for the approach, with docs and tests. Integrating it with each exporter will come in a separate PR if we think this is a good approach.

### CODE REVIEW

File: exporters/pelorus/config.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    # TODO: real logging
+    print(obj.__name__)
+
+    for field in fields(obj):
+        redact = field.metadata.get(__REDACT_KEY)
+        if redact is None and any(word in field.name.lower() for word in REDACT_WORDS):
+            redact = True
+
+        if redact:
```
Comment (KevinMGranger) [inline exporters/pelorus/config.py:33]: Double negatives are tricky. I should make this `log`.

**Comment by KevinMGranger** [comment]
I'm going to change parts of this to be based off of [attrs](https://attrs.org) instead. It'll make it more featureful while removing the amount of code in here, a simplifying some of this.

File: exporters/pelorus/__init__.py [hunk 2]
```diff
@@ -28,8 +28,12 @@ def _print_version():
...
     import __main__
 
-    # name of dir above app.py
-    exporter_name = pathlib.PurePath(__main__.__file__).parent.name
+    file = getattr(__main__, "__file__", None)
+    if file:
+        # name of dir above app.py
+        exporter_name = pathlib.PurePath(file).parent.name
+    else:
+        exporter_name = "INTERPRETER"
```
Comment (KevinMGranger) [inline exporters/pelorus/__init__.py:36]: This was necessary to get `import pelorus` working in a REPL. I can split this into a separate PR if wanted.

**Comment by KevinMGranger** [comment]
@mpryc let me know what you think. Good entry points to understanding this are:

[The module docs](https://github.com/konveyor/pelorus/pull/555/files#diff-857b478fcf7541910e5a9736005fb587664424a0f4d14b145ff902ac00d01842)
[The unit tests, to see more examples](https://github.com/konveyor/pelorus/pull/555/files#diff-55441f6aaa8d9b812c4b8dbf3e9bf0d62bc63c415214971b3c07dfe748f12648)
[DEVELOPING.md](https://github.com/konveyor/pelorus/pull/555/files#diff-3dc98b5b74bc0befceb386edeb3a9e10a50df1e4733d1beb313b8372e26a5eae) to understand more of the internals.

File: exporters/pelorus/config/DEVELOPING.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# How does this work?
```
Comment (weshayutin) [inline exporters/pelorus/config/DEVELOPING.md:1]: This seems like an odd way to begin the documentation. Perhaps start with some initial context vs. "this"

File: exporters/pelorus/config/__init__.py [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+    optional_list: list[str] = var(factory=list)
+
+    _private_field: Any = var(env_lookups=[])
+    public_but_ignore_me: Any = var(log=None)
+
+Notable differences:
+- anonymous_user will not be logged, when it otherwise would be.
+- should_pass_tests _will_ be logged.
+  We'd assume it's sensitive because it has "pass" in it, so we override that behavior.
+- whoami is set to the first value found of either `API_USER` or `USER`, instead of looking at `WHOAMI`.
```
Comment (weshayutin) [inline exporters/pelorus/config/__init__.py:78]: is or `USER` still true after https://github.com/konveyor/pelorus/pull/556 merges?

**Review by weshayutin — APPROVED** [review]
a few nits, nothing worth holding this back.. nice work!
I ran through and did a light test with it, everything passed.
I can run through some more scenarios later if needed.

File: exporters/pelorus/config/DEVELOPING.md (hunk 3 above)
Comment (KevinMGranger) [reply exporters/pelorus/config/DEVELOPING.md:1]: I figured that folks wouldn't _start_ with DEVELOPING.md, but additional context never hurts.

**Comment by weshayutin** [comment]
thanks for the udpates, lgtm . @mpryc last worked with you and should give it another pass.

**Comment by mpryc** [comment]
Thank you for awesome work. /lgtm

**Pull request merged**